scipy/spatial/ckdtree/src/query.cxx

```diff
@@ -381,6 +381,12 @@ query_single_point(const ckdtree *self,
                 npy_float64 tmp = x[split_dim] - split; 
                 if(NPY_LIKELY(p == 2)) {
                     inf2->side_distances()[split_dim] = tmp * tmp;
+                } else
+                if(NPY_LIKELY(p == 1)) {
+                    inf2->side_distances()[split_dim] = dabs(tmp);
+                } else
+                if(NPY_LIKELY(ckdtree_isinf(p))) {
```

Comment:
p == inf and p == 1 are conceptually different cases, even though it happens that they are doing the same thing in this part of the algorithm.

It's more of a styling choice. The compiler should be able to make a decision to merge them or not during optimization.